File: components/core/src/main/js/components/authentication/LogoutForm.js
import { withRouter, Link } from 'react-router-dom';
import ListErrors from '../ListErrors';
import React from 'react';
import { inject, observer } from 'mobx-react';

var loginPanel = {
 padding: '5rem 6rem 4rem 6rem',
 minHeight: 'auto'
};


@inject('authStore')
@withRouter
@observer
class LogoutForm extends React.Component {

  componentWillUnmount() {
    this.props.authStore.reset();
  }

  handleSubmitForm = (e) => {
    e.preventDefault();
    this.props.authStore.logout()
      .then(() => this.props.history.replace('/'));
  };

  render() {

    const { values, errors, inProgress } = this.props.authStore;


    return (

        <div className="container">

            <div className="col-md-2 hidden-sm hidden-xs">&nbsp;</div>

            <div className="col-md-8 col-xs-12">
                <h2>Logout</h2>

                <ListErrors errors={errors} />

                <form onSubmit={this.handleSubmitForm}>
                    <button className="btn btn-primary btn-sm" id="updateProfile" tabIndex="3" type="submit">
                        &nbsp; LOG OUT &nbsp;
                    </button>
                </form>
            </div>

            <div className="col-md-2 hidden-sm hidden-xs">&nbsp;</div>

        </div>
    );
  }
}

export default LogoutForm;

File: components/core/src/main/java/nextgen/templates/javascript/LogoutForm.java
package nextgen.templates.javascript;

public class LogoutForm {

	private final java.util.UUID uuid = java.util.UUID.randomUUID();
	private final org.stringtemplate.v4.STGroup stGroup;


	LogoutForm(org.stringtemplate.v4.STGroup stGroup) {
		this.stGroup = stGroup;
	}

	@Deprecated
	public java.util.UUID uuid() {
		return uuid;
	}

	public java.util.UUID getUuid() {
		return uuid;
	}

	@Override
	public String toString() {
		final org.stringtemplate.v4.ST st = stGroup.getInstanceOf("LogoutForm");
		return st.render().trim();
	}




	@Override
	public boolean equals(Object o) {
		if (this == o) return true;
		if (o == null || getClass() != o.getClass()) return false;
		LogoutForm that = (LogoutForm) o;
		return uuid.equals(that.uuid);
	}

	@Override
	public int hashCode() {
		return java.util.Objects.hash(uuid);
	}

	static final String st = "LogoutForm() ::= <<import { withRouter } from 'react-router-dom';\n" + 
				"import React from 'react';\n" + 
				"\n" + 
				"import { inject, observer } from 'mobx-react';\n" + 
				"\n" + 
				"import SignOut from '../components/SignOut.js';\n" + 
				"\n" + 
				"@withRouter\n" + 
				"@inject('authStore')\n" + 
				"@observer\n" + 
				"class LogoutForm extends React.Component {\n" + 
				"\n" + 
				"	constructor(){\n" + 
				"		super();\n" + 
				"		this.handleSubmitForm = this.handleSubmitForm.bind(this);\n" + 
				"	}\n" + 
				"\n" + 
				"	handleSubmitForm = (e) => {\n" + 
				"		e.preventDefault();\n" + 
				"		this.props.authStore.logout().then(() => this.props.history.replace('/')).catch(e => {});\n" + 
				"	}\n" + 
				"\n" + 
				"	render() {\n" + 
				"		return (<SignOut authStore={ this.props.authStore } onSubmit={ this.handleSubmitForm } />);\n" + 
				"	}\n" + 
				"}\n" + 
				"\n" + 
				"export default (LogoutForm); >>";
}  